src/test/java/com/github/dockerjava/cmd/PushImageCmdIT.java

```diff
@@ -50,30 +47,26 @@ public void pushLatest() throws Exception {
         String imageId = dockerRule.getClient().commitCmd(container.getId()).withRepository(username + "/busybox").exec();
 
         // we have to block until image is pushed
-        dockerRule.getClient().pushImageCmd(username + "/busybox").exec(new PushImageResultCallback()).awaitSuccess();
+        dockerRule.getClient().pushImageCmd(username + "/busybox").exec(new PushImageResultCallback()).awaitCompletion();
```

Comment:
it may complete with not pushed r tests needs awaitSuccess() everywhere as clear success of operation.

Reply:
I think that it is not a good choice to have an awaitComplete and an awaitSuccess. Since awaitComplete does not validate all error cases on pull and push

Reply:
Then it will hide bugs in .awaitSuccess logic. In any case success should be success for test even the next code do single checks of success criteria.